Sources/Datadog/Datadog.swift

```diff
@@ -218,7 +218,7 @@ public class Datadog {
         var rumInstrumentation: RUMInstrumentation?
 
         if let rumConfiguration = configuration.rum {
-            core.telemetry = RUMTelemetry(
+            DD.telemetry = RUMTelemetry(
```

Comment:
Good point 👍, I think we forgot about multiple instances of the SDK @maxep (well, AFAIR we discussed `core` <> `telemetry` relationship some time ago, but not recently when deciding on shift to `DD.telemetry`).

To some extent it isn't important which instance of the SDK uploads the telemetry from a running process. After all, telemetry will land in the same (Telemetry) Org and will consume the same user quota. Telemetry events by their nature are anonymous and self-contained, so I can't imagine situation where this would become problematic (any thoughts @fuzzybinary @maxep @xgouchet ?).

The only instance-specific information is the RUM context we inject into telemetry events:

https://github.com/DataDog/dd-sdk-ios/blob/f330ac00db328e110ca5287818c58cc9aaa706ec/Sources/Datadog/RUM/RUMTelemetry.swift#L74-L76

It will differ (or become wrong) based on initialization order if `DD.telemetry` is global.

Reply:
Also, maybe this is something we should decide on later, when the codebase is closer to its final V2 shape. Now we're clearly lacking utilities and `core` reference availability in many places, hence the `telemetry: Telemetry?` propagation is pretty tedious. I can imagine this will be simpler in few sprints from now (right before splitting into modules). WDYT @maxep ?

Reply:
Indeed, we had a brief discussion and concluded that a telemetry event could be dispatched on any user org. The idea was to register multiple cores to the global telemetry and let it balance events on all of them. But indeed, we have `core`-specific event data, e.g. [RUM Context](https://github.com/DataDog/dd-sdk-ios/blob/develop/Sources/Datadog/RUM/RUMTelemetry.swift#L110L112) and [App ID](https://github.com/DataDog/dd-sdk-ios/blob/develop/Sources/Datadog/RUM/RUMTelemetry.swift#L117).
Injecting `core` everywhere can be problematic (retain circle), especially in low-levels (like data persistency and uploads) and I don't think we need RUM data on every telemetry events 🤔 so it could be on a case by case: either the telemetry uses a specific `core`, or it can pick a random one from the registry.

As for now, multiple calls to init will override the previous SDK instance (not only the telemetry). We don't yet support multiple instances of the SDK so I think we can move forward with the `DD.telemetry` and tackle that when we commit to multiple instance support. How does it sounds?

Reply:
> We don't yet support multiple instances of the SDK so I think we can move forward with the DD.telemetry and tackle that when we commit to multiple instance support. How does it sounds?

Sounds good for me 👍. From the experience on changing V1 into V2 it seems very pragmatic to incur this small tech debt now and pay it off later when the codebase will be closer to its final shape. Especially, that we may need the same for `DD.logger` (it has instance-specific capability of `.verbosity` threshold), tackling it once at whole makes more sense.